model-governance/model_governance_graph.ipynb

```diff
@@ -0,0 +1,520 @@
+    "                fr=lookup_dict[asso['SourceArn']]\n",
+    "                to=lookup_dict[asso['DestinationArn']]\n",
+    "                if 'AssociationType' in asso:\n",
+    "                    label = str(asso['AssociationType'])\n",
+    "                else:\n",
+    "                    label = ''\n",
+    "                line = [idd, fr, to, label]\n",
+    "                f.write(','.join(map(str, line)) + '\\n')                \n",
+    "                #Sleep every 1000 calls to avoid throttle exception\n",
+    "                if int(ct) % 1000 == 0:\n",
+    "                    time.sleep(2) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "blank-perry",
+   "metadata": {},
+   "source": [
+    "## Upload the generated files to S3 \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "electoral-poland",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "BUCKET_NAME=''\n",
+    "!aws s3 cp nodes.csv nodes-tcs.csv nodes-actions.csv edges.csv s3://$BUCKET_NAME"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "given-designation",
+   "metadata": {},
+   "source": [
+    "## Setup Neptune Cluster and notebook plugin\n",
+    "\n",
+    "Create Neptune cluster either from the public CloudFormation Template or manually from the console. After this cell is complete, restart the notebook's kernel.\n",
+    "\n",
+    "You can create the Neptune cluster from [this](https://docs.aws.amazon.com/neptune/latest/userguide/machine-learning.html) CloudFormation template\n",
+    "\n",
+    "Also, the instructions to install the Graph notebook jupyter extension to visualize the graph from [HERE](https://github.com/aws/graph-notebook/#gremlin-server) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "social-submission",
+   "metadata": {},
+   "source": [
+    "## Setup Neptune IAM\n",
+    "Follow the instructions here: https://docs.aws.amazon.com/neptune/latest/userguide/iam-auth-enable.html"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "standing-survivor",
+   "metadata": {},
+   "source": [
+    "## Change Graph config and point to Neptune Endpoint\n",
+    "\n",
+    "If you get an error message: \"UsageError: Cell magic `%%graph_notebook_config` not found.\" use Jupyter Notebooks instead of jupyterlab and make sure that you have already installed the Graph extension from [HERE](https://github.com/aws/graph-notebook/#gremlin-server)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "medieval-canberra",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%graph_notebook_config\n",
+    "{\n",
+    "  \"host\": \"localhost\", ##<Change this value to point to Neptune Cluster.\n",
+    "  \"port\": 8182,\n",
+    "  \"auth_mode\": \"DEFAULT\",\n",
+    "  \"iam_credentials_provider_type\": \"ROLE\",\n",
+    "  \"load_from_s3_arn\": \"\",\n",
+    "  \"ssl\": false,\n",
+    "  \"aws_region\": \"us-east-1\"\n",
```

Comment:
Would this make this notebook only work in us-east-1?

Reply:
Users need to change this value with their relative regions as this is just a placeholder for an example